File: furuya.f90
program fizzbuzz
	integer :: i
	do i = 1, 100
		if (mod(i, 15) == 0) then
			print *, 'fizzbuzz'
		else if (mod(i, 3) == 0) then
			print *, 'fizz'
		else if (mod(i, 5) == 0) then
			print *, 'buzz'
		else 
			print *, i
		end if
	end do
end program fizzbuzz

File: furuya.pl
# ?- fizzbuzz.

transform(N, fizzbuzz) :- transform(N, fizz), transform(N, buzz).
transform(N, fizz) :- 0 is N mod 3.
transform(N, buzz) :- 0 is N mod 5.
transform(N, N).

do_fizzbuzz(101) :- !.
do_fizzbuzz(N) :-
	transform(N, Result), write(Result), nl,
	N1 is N + 1,
	do_fizzbuzz(N1).

fizzbuzz :- do_fizzbuzz(1).